packages/next/server/render.tsx

```diff
@@ -505,11 +502,14 @@ export async function renderToHTML(
     return ''
   }
 
-  let { renderServerComponentData } = renderOpts
-  if (isServerComponent && query.__flight__) {
-    renderServerComponentData = true
-    delete query.__flight__
-  }
+  let renderServerComponentData = isServerComponent
+    ? query.__flight__ !== undefined
+    : false
+
+  const serverComponentProps =
+    isServerComponent && query.__props__
+      ? JSON.parse(query.__props__ as string)
+      : undefined
```

Comment:
Looks good! Let's also `delete` both props (`__flight__` and `__props__`) so the component won't receive these queries.